remote.rb

```diff
@@ -3,24 +3,151 @@
+    nature:             elements_arr[elements_arr.index("Produsul")+1],
+    transactions:       transactions,
+    other_transactions: other_transactions
+  )
+end
+
+def trans_page (arr_1, arr_2, arr_3, page)
+  page.css('tbody tr').each do |element|
+    selector = element.css('td font tt')
+    if element.text.strip.match(/^\d{2}\/\d{2}\/\d{2}\s*[A-z]\S*/)
+      arr_1 << element
+    elsif element.text.strip.match(/^\d{2}\/\d{2}\/\d{2}\s{1,5}\d{2}\/\d{2}\/\d{2}/)
+      arr_2 << element
+    elsif element.text.strip.match(/^\d{2}\/\d{2}\/\d{2}$/)
+      arr_3 << selector.text.strip.gsub(/\"/,'')
+    end   
+  end 
+end
+
+DATE_DMY    = '%d/%m/%y'
+data_arr    = []
+itter       = 1
+index       = 0
+post_d      = nil
+day         = "owwb-ws-calendar-current-month-day"
+from_date   = "USER_PROPERTY_DATE_FROM_DATEPICKER_SWITCH"
+nav_prev    = "owwb-ws-calendar-nav-prev"
+trans_entry = "owwb_cs_submit_emulator form_owwb_ws_entryAccountDoing-"
+confirm     = "owwb-cs-default-button-bg"
+back_main   = "form_GO_TO_PARENT_SECTION"
+
+browser = Watir::Browser.new(:firefox)
+browser.goto 'https://wb.micb.md/frontend/auth/userlogin?execution=e2s1'
+
+until browser.link(:name => "main_menu_CP_HISTORY").exist?
+  print "Login: "
+  login    = gets.chomp
+  print "Password "
+  password = gets.chomp
+  browser.text_field(:name => "Login").when_present.set(login)
+  browser.text_field(:name => "password").when_present.set(password)
+  browser.link(:class => "owwb-cs-default-button").click
+end
+
+browser.link(:class => "owwb-cs-default-button").wait_while_present
 
 page = Nokogiri::HTML(browser.html)
 
-data = []
+page.css('.owwb-cs-slide-list-account-item').each do |item|
+  elements_arr = []
+
+  item.css('li').each do |li|  
+    li.children.children.children.each { |i| elements_arr << i.text.strip.gsub(/\s{2,100}/, '') } 
+  end 
+
+  elements_arr = elements_arr.select { |i| i =~ /[a-z, 1-8]/ }
+  
+  browser.link(:class => trans_entry + itter.to_s + "_1").click
+  browser.link(:class => trans_entry + itter.to_s + "_1").wait_while_present
+  browser.link(:id => from_date).wait_until_present
+  browser.link(:id => from_date).click
+  2.times do 
+    browser.link(:class => nav_prev).wait_until_present
+    browser.link(:class => nav_prev).click
+  end
+  browser.link(:class => day).wait_until_present  
+  browser.link(:class => day).click
+  browser.link(:class => confirm).click
+  browser.link(:class => confirm).wait_while_present
 
-page.css('.owwb-cs-slide-list-account-item').each do |i|
-    account  = i.css('.contract-number').text
-    balance  = i.css('.owwb-cs-slide-list-amount-value').text
-    currency = i.css('.owwb-cs-slide-list-amount-currency').text
+  itter += 1
 
-data.push(  
-    name:     account,
-    balance:  balance,
-    currency: currency
-)  
-end  
+  transactions_page = Nokogiri::HTML(browser.html)
 
-puts JSON.pretty_generate(data)
+  tr_arr_1       = []
```

Comment:
collect all these variable in to a hash like hash = { tr_arr_1: [], etc... } and pass it to method instead of all these params. it's not good when method receives 3-4-5 params.